Set "rolePrefix" in ReactiveMethodSecurityConfiguration

ttddyy opened the pull request:
Currently, `GrantedAuthorityDefaults` is not considered in
`ReactiveMethodSecurityConfiguration`.
This commit updates the configuration to be aware of
`GrantedAuthorityDefaults` and update `rolePrefix` when the bean is
available.

Also, use the same instance of `DefaultMethodSecurityExpressionHandler`
when constructing `ExpressionBasedAnnotationAttributeFactory`.

<!--
For Security Vulnerabilities, please use https://pivotal.io/security#reporting
-->

<!--
Thanks for contributing to Spring Security. Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #).
-->

jzheaux commented:
Thanks, @ttddyy! Would you also be able to write some tests that confirm that this behavior works as expected? I think none have been added to this point, but you should be able to do something similar to [`GlobalMethodSecurityConfigurationTests`](https://github.com/spring-projects/spring-security/blob/master/config/src/test/java/org/springframework/security/config/annotation/method/configuration/GlobalMethodSecurityConfigurationTests.java).

And then, is there any reason that you chose to have the  `@Bean` method depend on `DefaultMethodSecurityExpressionHandler` instead of `MethodSecurityExpressionHandler`?

ttddyy commented:
Hi @jzheaux 

Updated the PR for
- Added test - `ReactiveMethodSecurityConfigurationTests`
- Changed to take `MethodSecurityExpressionHandler` on `methodMetadataSource` bean

Thanks,

rwinch commented on config/src/main/java/org/springframework/security/config/annotation/method/configuration/ReactiveMethodSecurityConfiguration.java:
```diff
@@ -28,17 +29,22 @@
 	private int advisorOrder;
 
+	@Autowired(required = false)
+	private GrantedAuthorityDefaults grantedAuthorityDefaults;
```
Please add a package private setter and move the `@Autowired(required = false)` on to the setter to align with gh-6624

ttddyy replied:
@rwinch Updated the PR to add a package private setter for Spring Fu support.
